### PR TITLE
feat: Add .NETStandard2.1 target and downgrade Microsoft.Extensions.Caching.Memory

### LINKED ISSUE #98: Add netstandard2.1 and correct dependencies to match the standards
We are using CRQSlite in a .NET Core 2 project. It has been some time since we updated this package, but now we want to use the latest version. However, when we upgraded to 1.31.1 we ran into problems like `Could not load type 'Microsoft.Extensions.Logging.Abstractions.Internal.NullScope' from assembly 'Microsoft.Extensions.Logging.Abstractions, Version=3.1.3.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'.`.

We suspect that the problem is that the `netstandard2.0` target depends on `Microsoft.Extensions.Caching.Memory` version `3.1.3`. Version 3 of Microsoft.Extensions... packages means that this package is for .NET Core 3 as the major version number of the packages should match. We use other packages that depends on other `Microsoft.Extensions....` packages in the latest version 2, and I suspect that this it is this mixup in versions that make our program fail at runtime - e.g. `NullScope` was removed from version 2... to version 3..., so even if we don't use `NullScope` we still fail. We could solve this by taking an explicit dependecy on `Microsoft.Extensions.Logging.Abstractions` version 3.1.3.0 in our project, but as far as I can tell from different sources this is not advisable.

Would you be open for a PR which:
* adds a new netstandard2.1 target which will use `Microsoft.Extensions.Caching.Memory` version `3.1.3` (or 3.1.4 which is now the latest)
* downgrades `Microsoft.Extensions.Caching.Memory` in the netstandard2.0 target, so that it is possible to use in netstandard2.0 projects without running into dependency problems

Some related issues:
* https://stackoverflow.com/questions/56783260/net-standard-2-0-which-microsoft-extensions-dependencyinjection-abstractions
* https://stackoverflow.com/questions/58193023/microsoft-extensions-version-confusion
* https://github.com/serilog/serilog-extensions-logging/issues/158

### PR DESCRIPTION
**Pull request by mikkelbu**
Also reintroduced net452 target that was accidentally removed.

Fixes #98

### CODE REVIEW

File: Framework/CQRSlite.Tests/CQRSlite.Tests.csproj [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
   <PropertyGroup>    
-    <TargetFrameworks Condition=" '$(OS)' == 'Windows_NT' ">netcoreapp3.0;netcoreapp2.2;net471</TargetFrameworks>
```
Comment (mikkelbu) [inline Framework/CQRSlite.Tests/CQRSlite.Tests.csproj:3]: netcoreapp2.2 is no longer supported

File: Framework/CQRSlite/Infrastructure/InterfaceMap.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ internal static class InterfaceMap
         internal static string GetImplementationNameOfInterfaceMethod(this Type implementation, Type @interface,
             string methodName, params Type[] argtypes)
         {
-#if NET452 || NETSTANDARD2_0
+#if NET452 || NETSTANDARD2_0 || NETSTANDARD2_1
```
Comment (mikkelbu) [inline Framework/CQRSlite/Infrastructure/InterfaceMap.cs:11]: Perhaps we should revert this if, so one does not have to keep adding targets

Comment (gautema) [reply Framework/CQRSlite/Infrastructure/InterfaceMap.cs:11]: That sounds like a good idea. Will you do it? If not, I plan on merging and testing this evening and can do it then.

Comment (mikkelbu) [reply Framework/CQRSlite/Infrastructure/InterfaceMap.cs]: I've done it :)

**Pull request merged**